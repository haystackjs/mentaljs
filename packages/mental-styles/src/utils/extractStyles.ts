import { prepareStyles } from "./prepareStyles";
import { extractModifier } from "./extractModifier";

export function extractStyles(src: any) {
    let normal = prepareStyles(src);
    let hover = prepareStyles(extractModifier('hover', src));
    return {
        ...normal,
        '&:hover, &:focus': hover
    }
}