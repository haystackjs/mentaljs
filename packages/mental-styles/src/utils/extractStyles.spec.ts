import { extractStyles } from "./extractStyles";

describe('extractStyles', () => {
    it('should extract styles', () => {
        expect(extractStyles({
            color: 'red',
            hoverColor: 'blue'
        })).toMatchSnapshot();
    });
});