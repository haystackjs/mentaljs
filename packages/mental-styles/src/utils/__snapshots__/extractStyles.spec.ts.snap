// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractStyles should extract styles 1`] = `
Object {
  "&:hover, &:focus": Object {
    "color": "blue",
  },
  "color": "red",
}
`;
